benchbuild/utils/slurm.py

```diff
@@ -111,21 +129,19 @@ def __save__(script_name, benchbuild, experiment, projects):
                 clean_lockfile=str(CFG["slurm"]["node_dir"]) + \
                     ".clean-in-progress.lock",
                 cpus=int(CFG['slurm']['cpus_per_task']),
-                exclusive=bool(CFG['slurm']['exclusive']),
                 lockfile=str(CFG['slurm']["node_dir"]) + ".lock",
                 log=logs_dir.resolve() / str(experiment.id),
                 max_running=int(CFG['slurm']['max_running']),
                 name=experiment.name,
-                nice=int(CFG['slurm']['nice']),
                 nice_clean=int(CFG["slurm"]["nice_clean"]),
                 node_command=node_command,
-                no_multithreading=not CFG['slurm']['multithread'],
                 ntasks=1,
                 prefix=str(CFG["slurm"]["node_dir"]),
                 projects=projects,
                 slurm_account=str(CFG["slurm"]["account"]),
                 slurm_partition=str(CFG["slurm"]["partition"]),
-                timelimit=str(CFG['slurm']['timelimit']),
+                sbatch_options='\n'.join(
+                    [s_opt.to_slurm_opt() for s_opt in slurm_options]),
```

Comment:
I think we can get rid of this by adding a small function to the slurm specific requirements code.

```{python}
def slurm_requirement(input: Requirement) -> SlurmRequirement:
  if isinstance(input, SlurmRequirement):
    return tp.cast(SlurmRequirement, input)
  else:
    raise NotImplementedError(...)
```

Then we can make it explicit that all requirements that we use here need to be Slurm-Specific.

Reply:
No Idea why gh needs 4 ever to update this but the is already a commit that fixes this